src/Symfony/Component/HttpFoundation/File/MimeType/MimeTypeByExtensionGuesser.php

```diff
@@ -0,0 +1,1023 @@
+        'hh' => 'text/x-c',
+        'dic' => 'text/x-c',
+        'f' => 'text/x-fortran',
+        'for' => 'text/x-fortran',
+        'f77' => 'text/x-fortran',
+        'f90' => 'text/x-fortran',
+        'java' => 'text/x-java-source',
+        'nfo' => 'text/x-nfo',
+        'opml' => 'text/x-opml',
+        'p' => 'text/x-pascal',
+        'pas' => 'text/x-pascal',
+        'etx' => 'text/x-setext',
+        'sfv' => 'text/x-sfv',
+        'uu' => 'text/x-uuencode',
+        'vcs' => 'text/x-vcalendar',
+        'vcf' => 'text/x-vcard',
+        '3gp' => 'video/3gpp',
+        '3g2' => 'video/3gpp2',
+        'h261' => 'video/h261',
+        'h263' => 'video/h263',
+        'h264' => 'video/h264',
+        'jpgv' => 'video/jpeg',
+        'jpm' => 'video/jpm',
+        'jpgm' => 'video/jpm',
+        'mj2' => 'video/mj2',
+        'mjp2' => 'video/mj2',
+        'mp4' => 'video/mp4',
+        'mp4v' => 'video/mp4',
+        'mpg4' => 'video/mp4',
+        'mpeg' => 'video/mpeg',
+        'mpg' => 'video/mpeg',
+        'mpe' => 'video/mpeg',
+        'm1v' => 'video/mpeg',
+        'm2v' => 'video/mpeg',
+        'ogv' => 'video/ogg',
+        'qt' => 'video/quicktime',
+        'mov' => 'video/quicktime',
+        'uvh' => 'video/vnd.dece.hd',
+        'uvvh' => 'video/vnd.dece.hd',
+        'uvm' => 'video/vnd.dece.mobile',
+        'uvvm' => 'video/vnd.dece.mobile',
+        'uvp' => 'video/vnd.dece.pd',
+        'uvvp' => 'video/vnd.dece.pd',
+        'uvs' => 'video/vnd.dece.sd',
+        'uvvs' => 'video/vnd.dece.sd',
+        'uvv' => 'video/vnd.dece.video',
+        'uvvv' => 'video/vnd.dece.video',
+        'dvb' => 'video/vnd.dvb.file',
+        'fvt' => 'video/vnd.fvt',
+        'mxu' => 'video/vnd.mpegurl',
+        'm4u' => 'video/vnd.mpegurl',
+        'pyv' => 'video/vnd.ms-playready.media.pyv',
+        'uvu' => 'video/vnd.uvvu.mp4',
+        'uvvu' => 'video/vnd.uvvu.mp4',
+        'viv' => 'video/vnd.vivo',
+        'webm' => 'video/webm',
+        'f4v' => 'video/x-f4v',
+        'fli' => 'video/x-fli',
+        'flv' => 'video/x-flv',
+        'm4v' => 'video/x-m4v',
+        'mkv' => 'video/x-matroska',
+        'mk3d' => 'video/x-matroska',
+        'mks' => 'video/x-matroska',
+        'mng' => 'video/x-mng',
+        'asf' => 'video/x-ms-asf',
+        'asx' => 'video/x-ms-asf',
+        'vob' => 'video/x-ms-vob',
+        'wm' => 'video/x-ms-wm',
+        'wmv' => 'video/x-ms-wmv',
+        'wmx' => 'video/x-ms-wmx',
+        'wvx' => 'video/x-ms-wvx',
+        'avi' => 'video/x-msvideo',
+        'movie' => 'video/x-sgi-movie',
+        'smv' => 'video/x-smv',
+        'ice' => 'x-conference/x-cooltalk',
+    );
+
+    /**
+     * @param string $extension The extension of a file (i.e. 'jpg' or 'uvp')
+     *
+     * @return string|null
+     */
+    public function guess($extension)
+    {
+        return isset($this->defaultExtensions[$extension]) ? $this->defaultExtensions[$extension] : null;
```

Comment:
I don't know if it worth it, but PNG, JPEG, etc are valid extensions, what about lowercase the extension before use it?